pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -567,14 +567,19 @@ func (ctrl *Controller) syncImageConfig(key string) error {
 			isNotFound := errors.IsNotFound(err)
 			rci := createNewRegistriesConfigIgnition(registriesTOML)
 			if !isNotFound && equality.Semantic.DeepEqual(rci, mc.Spec.Config) {
-				applied = false
-				return nil
+				// if the configuration for the registries is equal, we still need to compare
+				// the generated controller version because during an upgrade we need a new one
+				mcCtrlVersion := mc.Annotations[ctrlcommon.GeneratedByControllerVersionAnnotationKey]
+				if mcCtrlVersion == version.Version.String() {
+					applied = false
+					return nil
+				}
```

Comment:
I guess we can also just update that annotation to the new version, but let's keep it this way for now